### PR TITLE
added goal visualization to planner_manager

### PR DESCRIPTION
**Pull request by abajcsy**
Changed the unused bound_pub to be goal_pub and publish a marker at the goal state upon MaybeRequestTrajectory() call. Updated the position-velocity manager launch file to reflect the change of bound_pub to goal_pub (and corresponding topics).

### CODE REVIEW

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h [hunk 1]
```diff
@@ -97,6 +97,9 @@ class PlannerManager : private Uncopyable {
   // derived classes should still call thhis function.
   virtual void TimerCallback(const ros::TimerEvent& e);
 
+  // Create and publish a marker at goal state. 
+  virtual void VisualizeGoal();
```
Comment (dfridovi) [inline ros/src/fastrack/include/fastrack/planning/planner_manager.h]: Make this function `const`.

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h [hunk 2]
```diff
@@ -330,6 +336,46 @@ void PlannerManager<S>::TimerCallback(const ros::TimerEvent& e) {
   tf_broadcaster_.sendTransform(tf);
 }
 
+// Converts the goal state into a Rviz marker. 
+template<typename S>
+void PlannerManager<S>::VisualizeGoal() {
+  // Set up sphere marker.
+  visualization_msgs::Marker sphere;
+  sphere.ns = "sphere";
+  sphere.header.frame_id = this->fixed_frame_;
```
Comment (dfridovi) [inline ros/src/fastrack/include/fastrack/planning/planner_manager.h]: Don't need `this->` here. You only need that when you're a templated derived class from a templated base and you're trying to use a member variable of your templated base.

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h [hunk 3]
```diff
@@ -330,6 +336,46 @@ void PlannerManager<S>::TimerCallback(const ros::TimerEvent& e) {
...
+  sphere.color.a = 1.0;
+  sphere.color.r = 0.3;
+  sphere.color.g = 0.7;
+  sphere.color.b = 0.7;
+
+  geometry_msgs::Point center;
+
+  // Fill in center and scale.
+  sphere.scale.x = 0.1;
+  std::cout << "goal at: " << goal_ << std::endl;
```
Comment (dfridovi) [inline ros/src/fastrack/include/fastrack/planning/planner_manager.h]: Please don't pollute `stdout`.

**Review by dfridovi — CHANGES_REQUESTED** [review]
You also need to change `software_demo.launch` where we launch the planner manager to rename the appropriate param and arg.

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h (hunk 1 above)
Comment (abajcsy) [reply ros/src/fastrack/include/fastrack/planning/planner_manager.h]: fixed

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h (hunk 2 above)
Comment (abajcsy) [reply ros/src/fastrack/include/fastrack/planning/planner_manager.h]: fixed

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h (hunk 3 above)
Comment (abajcsy) [reply ros/src/fastrack/include/fastrack/planning/planner_manager.h]: fixed.

**Comment by abajcsy** [comment]
Updated ```software_demo.launch``` with the param and args and removed prints and made changes to ```VisualizeGoal()``` function.

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h [hunk 4]
```diff
@@ -98,7 +98,7 @@ class PlannerManager : private Uncopyable {
   virtual void TimerCallback(const ros::TimerEvent& e);
 
   // Create and publish a marker at goal state. 
-  virtual void VisualizeGoal();
+  const virtual void VisualizeGoal();
```
Comment (dfridovi) [inline ros/src/fastrack/include/fastrack/planning/planner_manager.h]: lolz nooooooo! I meant `virtual void VisualizeGoal() const;`

**Review by dfridovi — CHANGES_REQUESTED** [review]
Look at `VisualizeGoal`

File: ros/src/fastrack/include/fastrack/planning/planner_manager.h (hunk 4 above)
Comment (abajcsy) [reply ros/src/fastrack/include/fastrack/planning/planner_manager.h]: lmao sorry -- will change!

Comment (dfridovi) [reply ros/src/fastrack/include/fastrack/planning/planner_manager.h]: Also change the implementation signature to match.

**Comment by abajcsy** [comment]
Fixed const issue.

**Review by dfridovi — APPROVED** [review]
LGTM! Approved if software demo runs.

**Comment by abajcsy** [comment]
Software demo runs!

**Pull request merged**